File: languages/php/Bloock/Entity/IdentityV2/DecimalAttributeDescriptor.php
<?php

namespace Bloock\Entity\IdentityV2;

use Bloock\DecimalAttributeDefinitionV2;

class DecimalAttributeDescriptor extends AttributeDescriptor
{
    public static function fromProto(DecimalAttributeDefinitionV2 $res): DecimalAttributeDescriptor
    {
        return new DecimalAttributeDescriptor($res->getDisplayName(), $res->getId(), $res->getDescription(), $res->getRequired());
    }

    public function toProto(): DecimalAttributeDefinitionV2
    {
        $p = new DecimalAttributeDefinitionV2();
        $p->setDisplayName($this->displayName);
        $p->setId($this->technicalName);
        $p->setDescription($this->description);
        $p->setRequired($this->required);
        return $p;
    }
}

File: languages/java/src/main/java/com/bloock/sdk/entity/identity_v2/DecimalAttributeDescriptor.java
package com.bloock.sdk.entity.identity_v2;

import com.bloock.sdk.bridge.proto.IdentityEntitiesV2;

/**
 * Represents a descriptor for an attribute with a decimal value.
 */
public class DecimalAttributeDescriptor extends AttributeDescriptor {
  /**
   * Constructs an DecimalAttributeDescriptor object with the specified parameters.
   * @param displayName
   * @param technicalName
   * @param description
   * @param required
   */
  public DecimalAttributeDescriptor(
      String displayName, String technicalName, String description, Boolean required) {
    super(displayName, technicalName, description, required);
  }

  public static DecimalAttributeDescriptor fromProto(
      IdentityEntitiesV2.DecimalAttributeDefinitionV2 res) {
    return new DecimalAttributeDescriptor(
        res.getDisplayName(), res.getId(), res.getDescription(), res.getRequired());
  }

  public IdentityEntitiesV2.DecimalAttributeDefinitionV2 toProto() {
    return IdentityEntitiesV2.DecimalAttributeDefinitionV2.newBuilder()
        .setDisplayName(this.displayName)
        .setId(this.technicalName)
        .setDescription(this.description)
        .setRequired(this.required)
        .build();
  }
}
